docs/modules/general/pages/architecture-overview/mfe.adoc

```diff
@@ -1,37 +1,40 @@
 
 == Why Microfrontends
-To precisely explain the purpose of using Microfrontends, it is crucial to become familiar with the client side of each Microfrontend.
+To precisely explain the purpose of using Microfrontends, it is crucial to become familiar with the client side perspective of each Microfrontend.
 
 === Microfrontend clients
-As previously stated, each Microfrontend is exposing a remote entry file, that can be accessed by any application. This remote entry file should contain all the data required to display the desired pieces of user interface. In the context of OneCX platform, this file is accessed and used in one of those situations:
-* **Module display**: Shell Application during workspace construction creates routing rules and associates a remote entry file with a specific route. Whenever user navigates to that route remote entry file is used (with other data of the current workspace) to display desired module in the content pane of the page.
-* **Remote Component display**: Any of Microfrontend's module, Remote Component or Shell Application in any period during platform runtime uses remote entry file to display a piece of user interface within a page.
+As previously stated, each Microfrontend is exposing a remote entry file, that can be accessed by any UI App. This remote entry file should contain all the data required to display the desired pieces of user interface. In the context of OneCX platform, this file should expose the following types of content:
 
-// TODO: Add link to remote components
-Module display is suited for whole UI modules, which consists of many pages and its own routing rules, whereas Remote Components are suited for smaller pieces of UI, which can be easily replacable and reusable within the platform. It is important to know that **Remote Components and Modules are both exposed via Microfrontend through a remote entry file**. More on the topic of Remote Components can be found here(TODO).
+* **Module**: Shell during workspace construction creates routing rules and associates a module from remote entry file with a specific route. Whenever user navigates to that route module is loaded using remote entry file to display desired content in the content pane of the page.
+* **Remote Component**: Component exposed via remote entry file is loaded onto a page on demand using Slot Component.
+
+Module type is suited for displaying whole UI modules containing many pages and its own routing rules, whereas Remote Component is suited for smaller pieces of UI, which can be easily replacable and reusable within the platform. It is important to know that **Remote Components and Modules are both exposed via a Microfrontend through a remote entry file**. More on the topic of Remote Components can be found xref:architecture-overview/remoteComponents.adoc[here].
```

Comment:
... , which ~~can be easily~~ are easily replaceable and reusable within the platform.
or
..., which can be easily ~~replacable and reusable~~ replaced and reused within the platform.